src/io.rs

```diff
@@ -97,4 +99,85 @@ impl Write for CallbackWriter {
     fn flush(&mut self) -> Result<()> {
         Ok(())
     }
+
+    fn write_vectored(&mut self, bufs: &[IoSlice<'_>]) -> Result<usize> {
+        let mut out_n: usize = 0;
+        for buf in bufs.iter().filter(|b| !b.is_empty()) {
+            out_n += match self.write(buf) {
+                Ok(n) => {
+                    if n != buf.len() {
+                        return Ok(out_n + n);
+                    }
+                    n
+                }
+                e => return e,
```

Comment:
I think this is not right. If we've already written 100 bytes, and then we get EWOULDBLOCK, we need to return Ok(100), not Error.